public/app/features/alerting/unified/components/mute-timings/MuteTimingForm.tsx

```diff
@@ -70,19 +78,51 @@ const MuteTimingForm = ({ muteTiming, showError, loading, provenance }: Props) =
 
     const newMuteTiming = createMuteTiming(values);
 
-    const muteTimings = muteTiming
-      ? config?.mute_time_intervals?.filter(({ name }) => name !== muteTiming.name)
-      : config?.mute_time_intervals;
-
+    const originalMuteTimingsWithoutNew = fromMuteTimings
```

Comment:
Can we write a separate function like `replaceMuteTiming` to handle the omit + re-add logic?